### ISSUE TITLE
AppCenterPostBuild.GetAppFilePath should use Application.productName

### ISSUE THREAD

**Issue by IS-GYG** [Microsoft/AppCenter-SDK-Unity#224]
Hi guys,

We recently imported the AppCenter SDK Unity into our project and could not build the UWP build. turns our the problem was coming from the follow method GetAppFilePath in AppCenterPostBuild.cs, specifically the line:
var candidate = Path.Combine(pathToBuiltProject, PlayerSettings.WSA.tileShortName);

This should be 
var candidate = Path.Combine(pathToBuiltProject, Application.productName);

The reason is because Unity will always use the productName to generate the sub folder in the output path, but the titleShortName is up to the developers discretion (in our case the Title Short name was different from the Product name and had spaces in it).

Also for the method AppCenterPostBuild.InjectCodeToFile, there should be a safe check for when GetAppFilePath fails for whatever reason and returns null:

I put this at the top of the function to avoid the null exception from File.ReadAllText during the build process:

if(string.IsNullOrEmpty(appFilePath))
        {
            // TODO Update documentation link
            Debug.LogError("Unable to automatically modify file '" + appFilePath + "'. For App Center Push to work properly, " +
                           "please follow troubleshooting instructions at https://docs.microsoft.com/en-us/mobile-center/sdk/troubleshooting/unity");

            return;
        }

If you guys can make me a contributor (or show me how, I can make a pull request with my changes).

Cheers,

**Comment by annakocheshkova** [comment]
Hi @YawJatah and thank you for this information, it was very helpful!

We have already faced with this issue recently and are in the process of fixing it. 
We are going to make a PR with these fixes and a new SDK release in a couple of days.
If you don't mind, we will add you as a reviewer. 

Thanks!

**Comment by IS-GYG** [comment]
Thanks for the quick response @annakocheshkova . Will await the new SDK with the fixes.

Cheers

**Comment by evgeny-pol** [comment]
Hello @YawJatah , we have fixed this bug in the latest version of the SDK, please test if it works for you:
https://github.com/Microsoft/AppCenter-SDK-Unity/releases/tag/1.2.1

I will close this issue for now, but do not hesitate to reopen it if it does not work or you have any other questions.

**Comment by IS-GYG** [comment]
Hi @evgeny-pol , I just updated the Appcenter SDK for Unity to 1.2.1, and the main issue with regards to the path is indeed resolved. But in the process there is what i believe is a logic bug with regards to code injection. It will manifest itself as a build failure if you have your build pipeline set to strict (we use the Unity command line to automate our build).

The issue in question is in the function AppCenterPostBuild.InjectCodeToFile, it has to the do with line 160

`if (matches.Success)`

If that regex fails then the else has a Debug.LogError (in strict mode a LogError is considered an error so the build process fail). The code itself is fine, it has to do with the reason for the failure, in our case it was because the injection code was already there, so the regex could not find the default code. I had to modify the code to take this scenario into account:

`
       var fileText = File.ReadAllText(appFilePath);

        if (fileText.Contains(commentText))
        {
            if (fileText.Contains(codeToInsert))
            {
                Debug.Log(string.Format("AppCenterPostBuild.InjectCodeToFile - Code file [{0}], already containts the injection code. Will not re-inject", appFileName));
            }
            else
            {
                Debug.LogWarning(string.Format("AppCenterPostBuild.InjectCodeToFile - Code file [{0}], already containts the injection code but it does not match the latest code injecttion. Plese refer to the code injection file {1}", appFileName, codeToInsertFileName));
            }
            
            return;
        }`

My modification checks to see if the code is already there and if is we just do a normal log. If the code doesn't match (if for whatever reason in the future the injection code changes), then we warn.

I know the issue is closed, but this bug is directly  related to the my previous reported issue.

Thanks for taking investigating this future.
